include/grids.php

```diff
@@ -203,4 +220,36 @@
     $req->execute(array($_POST['del_admin_id']));
   }
 
+  // ---------------- UPDATE CONFIG ----------------
+  else if(isset($_POST['update_config'])){
+
+      $pathinfo = pathinfo($_POST['config_file']);
+
+      $config_full_uri = $_POST['config_file']; // the complete path to the file, including the file (name) its self and the fully qualified path
+      $config_full_path = $pathinfo['dirname']; // path to file (without filename its self)
+      $config_name = basename($_POST['config_file']); // config file name only (without path)
+      $config_parent_dir = basename($config_full_path); // name of the dir that contains the config file (without path)
+
+      /*
+       * create backup for history
+       */
+      if (!file_exists($dir="../$config_full_path/history"))
+         mkdir($dir, 0777, true);
```

Comment:
Why do you need 777?

Reply:
indeed we do not need 777, my mistake - forgot that this dir will created with the webserver's user so indeed 700 should reach out